src/base-plugins/RadioGroupPlugin.js

```diff
@@ -0,0 +1,112 @@
+import { BasePlugin } from './BasePlugin';
+import { RadioGroup } from '../ui-elements/RadioGroup';
+
+/**
+ *
+ *
+ * @export
+ * @class RadioGroupPlugin
+ */
+export class RadioGroupPlugin extends BasePlugin {
+  /**
+   *
+   *Creates an instance of RadioGroupPlugin.
+   * @constructor
+   * @memberof RadioGroupPlugin
+   * @param {string} name
+   */
+  constructor(cssSelector, name, {supportedValues, initialValue, controlName, featureName, radioCount}) {
+    super(name);
+    this.selectors = cssSelector ? cssSelector.split(',') : [];
+    this.supportedValues = supportedValues;
+    this.initialValue = initialValue;
+    this.controlName = controlName;
+    this.featureName = featureName;
+    this.radioCount = radioCount;
+
+    this._currentValue = initialValue ? initialValue : this.supportedValues[0];
+
+    this.radioGroups = this.setUpRadios(this.selectors);
+
+    this.radioGroupsLength = this.radioGroups.length;
+  }
+
+  /**
+   * @memberof RadioGroupPlugin
+   * @param {string[]} selectors the separated selector strings used to target the radio button groups
+   * @returns {RadioGroup[]}
+   */
+  setUpRadios(selectors) {
+    const radioGroups = [];
+
+    selectors.forEach((selector) => {
```

Comment:
I'm curious if this would read better as a map with a filter:

```javascript
return selectors
  .map((selector) => {
    return new RadioGroup({ ... });
  })
  .filter((radioGroup) => {
    if (radioGroup.length !== this.radioCount) {
      this.warn('...');
      return false;
    } else if (!radioGroup.hasOnly(this.supportedValues)) {
      return false;
    } else if (radioGroup.hasDupicateValues()) {
      this.warn('...');
      return false;
    }
    return true;
  });
```